plugins/inputs/marklogic/marklogic.go

```diff
@@ -0,0 +1,233 @@
+		Meta                struct {
+			URI         string    `json:"uri"`
+			CurrentTime time.Time `json:"current-time"`
+			ElapsedTime struct {
+				Units string  `json:"units"`
+				Value float64 `json:"value"`
+			} `json:"elapsed-time"`
+		} `json:"meta"`
+		StatusProperties struct {
+			Online struct {
+				Units string `json:"units"`
+				Value bool   `json:"value"`
+			} `json:"online"`
+			LoadProperties struct {
+				TotalLoad struct {
+					Units string  `json:"units"`
+					Value float64 `json:"value"`
+				} `json:"total-load"`
+			} `json:"load-properties"`
+			StatusDetail struct {
+				HostMode struct {
+					Units string `json:"units"`
+					Value string `json:"value"`
+				} `json:"host-mode"`
+				Cpus struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"cpus"`
+				Cores struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"cores"`
+				CoreThreads struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"core-threads"`
+				TotalCPUStatUser   float64 `json:"total-cpu-stat-user"`
+				TotalCPUStatNice   float64 `json:"total-cpu-stat-nice"`
+				TotalCPUStatSystem float64 `json:"total-cpu-stat-system"`
+				TotalCPUStatIdle   float64 `json:"total-cpu-stat-idle"`
+				TotalCPUStatGuest  float64 `json:"total-cpu-stat-guest"`
+				MemoryProcessSize  struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-process-size"`
+				MemoryProcessRss struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-process-rss"`
+				MemorySystemTotal struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-system-total"`
+				MemorySystemFree struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"memory-system-free"`
+				DataDirSpace struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"data-dir-space"`
+				QueryReadBytes struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"query-read-bytes"`
+				QueryReadLoad struct {
+					Units string `json:"units"`
+					Value int    `json:"value"`
+				} `json:"query-read-load"`
+				HTTPServerReceiveBytes struct {
+					Units string  `json:"units"`
+					Value float64 `json:"value"`
+				} `json:"http-server-receive-bytes"`
+				HTTPServerSendBytes struct {
+					Units string  `json:"units"`
+					Value float64 `json:"value"`
+				} `json:"http-server-send-bytes"`
+			} `json:"status-detail"`
+		} `json:"status-properties"`
+	} `json:"host-status"`
+}
+
+func (c *Marklogic) Description() string {
+	return "Gathers host health status data from a Marklogic Cluster"
+}
+
+var sampleConfig = `
+  ## List URLs of Marklogic hosts using Management API endpoint.
+  # hosts = ["http://localhost:8002/manage/v2/hosts/${hostname}?view=status&format=json"]
```

Comment:
+1 for using /manage/v2/hosts to grab all hosts in the cluster.  It could maybe work like this:

- if a host is listed `http://localhost:8002/manage/v2/hosts/foobar` then only connect to it
- if no host is listed `http://localhost:8002/manage/v2/hosts` then do them all
- in either case, don't require the user to specify `?view=status&format=json`, you can do that automatically.

Reply:
Good idea Daniel, unfortunately their api only returns links to the host's information and not the information itself:

```json
{
  "hosts": [
    "http://localhost:8002/manage/v2/hosts/foobar"
  ]
}
```

Still very usable, but it doesn't provide a way to fetch all hosts in one call. After finding that out, we decided it was ok to collect individual listed hosts until we learn that there can be a great many hosts, at which point we'd switch to something like your recommendation.